app/models.py

```diff
@@ -0,0 +1,197 @@
+            current_app.config['SECRET_KEY'], expires_in=expiration)
+        return s.dumps({'user_id': self.user_id})
+
+    @staticmethod
+    def verify_auth_token(token):
+        # This function verifies the token supplied by the user
+        s = Serializer(current_app.config['SECRET_KEY'])
+        try:
+            data = s.loads(token)
+        except:
+            return None
+        return User.query.get(data['user_id'])
+
+    @password.setter
+    def password(self, password):
+        # This function sets the user password hash from the password supplied
+        self.password_hash = generate_password_hash(password)
+
+    def verify_password(self, password):
+        # This function verifies the user password against its hash
+        return check_password_hash(self.password_hash, password)
+
+    def save(self):
+        # This function save the user details to the database.
+        db.session.add(self)
+        db.session.commit()
+
+    def get_details(self):
+        # This function returns the details of the current user
+        user_details = {
+            'first_name': self.first_name,
+            'last_name': self.last_name,
+            'email': self.email,
+            'short_urls': [
+                url_for('api.shorturls', _external=True) +
+                str(short_url.short_url_id)
+                for short_url in self.short_urls
+            ]}
+        return user_details
+
+    def __repr__(self):
+        # This returns a machine representation of a user
+        return '<User %r>' % (self.first_name + " " + self.last_name)
+
+
+class ShortUrl(db.Model):
+    """
+    This is the short url which has a many to many relationship with the
+    Visitor model and a many to one to the User model.
+    """
+    __tablename__ = 'short_urls'
+    short_url_id = db.Column(db.Integer, primary_key=True)
+    short_url = db.Column(db.String(64), unique=True)
+    active = db.Column(db.Integer, index=True, default=1)
+    deleted = db.Column(db.Integer, default=0)
+    date_time = db.Column(db.DateTime, index=True, default=datetime.now())
+    user_id = db.Column(db.Integer, db.ForeignKey('users.user_id'))
+    long_url_id = db.Column(db.Integer, db.ForeignKey('long_urls.long_url_id'))
+
+    def save(self):
+        # This saves the short URL into the database
+        db.session.add(self)
+        db.session.commit()
+
+    def get_details(self):
+        # This returns the short url details as a json object
+        visit = db.session.query(ShortUrl, db.func.count(
+            visits.c.short_url_id)).outerjoin(visits).group_by(
+            ShortUrl.short_url_id).filter_by(
+            short_url_id=self.short_url_id).first()
+        no_of_visits = visit[1] if visit else 0
+        url_details = {
+            'short_url': self.short_url,
+            'short_url_url': url_for(
+                'api.shorturl', id=self.short_url_id, _external=True),
+            'long_url': self.long_url.long_url,
```

Comment:
Due to the many to one relationship this model have with LongUrl model, the relationship declare at L148 in the LongUrl model makes long_url available in the self.